src/components/10-atoms/carousel/index.js

```diff
@@ -109,57 +114,41 @@ class AXACarousel extends InlineStyles {
   }
 
   _setSlideVisibleWithAnimation(slideIndex, animationClass) {
-    this.visibleSlideIndex = slideIndex;
-    this.slides.forEach(node => {
-      node.style.display = 'none';
-    });
-    this.slides[slideIndex].style.display = 'block';
+    this.visible = slideIndex;
     this._animationWrapperClass = animationClass;
   }
 
-  _nextSlide() {
-    let nextSlideIndex = this.visibleSlideIndex + 1;
-
-    if (nextSlideIndex > this.slides.length - 1) {
-      nextSlideIndex = 0;
-    }
-
-    this._setSlideVisibleWithAnimation(nextSlideIndex, ANIMATION_RIGHT_CLASS);
+  _nextSlide(increment = 1) {
+    let numSlides = this.slides.length;
+    // wrap around after last slide...
+    let nextSlideIndex = (this.visible + increment) % numSlides;
+    this._setSlideVisibleWithAnimation(
+      nextSlideIndex < 0 ? numSlides - 1 : nextSlideIndex, // ... and before first slide
+      increment > 0 ? ANIMATION_RIGHT_CLASS : ANIMATION_LEFT_CLASS
+    );
```

Comment:
For me the old solution is better readable. There are just simple mathematical operations like ">" and less ifs.